### ISSUE TITLE
Convert sample to use BackgroundService

### ISSUE THREAD

**Issue by guardrex** [aspnet/Docs#7638]
Convert the *BackgroundTasksSample-WebHost* sample over to [BackgroundService](https://docs.microsoft.com/dotnet/api/microsoft.extensions.hosting.backgroundservice).

* Requires commensurate changes to the topic's text.
* Probably need to snapshot 2.0 code, since `BackgroundService` is new in 2.1.

See: https://github.com/aspnet/Docs/pull/7397

**Comment by granthoff1107** [comment]
Yeah let me know what I need to do

**Comment by guardrex** [comment]
Sure ...

1. In the snapshot folder for the WebHost version of the sample at ...

aspnetcore/fundamentals/host/hosted-services/samples-snapshot/2.x/BackgroundTasksSample-WebHost

... create a copy of the current *QueuedHostedService.cs* file ...

aspnetcore/fundamentals/host/hosted-services/samples/2.x/BackgroundTasksSample-WebHost/Services/QueuedHostedService.cs

... this will serve as a snapshot of the 2.0 version. We'll show it for now, since `BackgroundService` is only available for 2.1+.

2. Make the changes that you had over on #7397 to the file at ...

aspnetcore/fundamentals/host/hosted-services/samples/2.x/BackgroundTasksSample-WebHost/Services/QueuedHostedService.cs

... just as you did on your patch branch before.

3. In the topic, where the prior `QueuedHostedService.cs` was described for the 2.0 sample, you'll isolate existing text in a *moniker range* that will only show that text when the 2.0 version of the topic is loaded. When you provide alternate text that describes the `BackgroundService` version, that text will receive a *moniker range* for 2.1 or later. The links to the two sample files only differ by the word "snapshot" in a link segment. Here's an example ...

```
::: moniker range=">= aspnetcore-2.1"

Describe the new BackgroundService version here. To link to the 2.1 sample file, use this ...

[!code-csharp[](hosted-services/samples/2.x/BackgroundTasksSample-WebHost/Services/QueuedHostedService.cs?name=snippet1&highlight=30-31,35)]

::: moniker-end

::: moniker range="= aspnetcore-2.0"

Retain the current text from the topic for 2.0 readers. To link to the 2.0 sample file, use 
the following (note "snapshot" in the path segment). You just need to re-target the 
current link to the *samples-snapshot* folder ...

[!code-csharp[](hosted-services/samples-snapshot/2.x/BackgroundTasksSample-WebHost/Services/QueuedHostedService.cs?name=snippet1&highlight=30-31,35)]

::: moniker-end
```

The *moniker ranges* and build system will take care of the rest.

Note on sample links: You'll see we use *snippets* to isolate the code for the topic to display. You can *highlight* lines and ranges of lines as needed.

Since you'll be working with multiple files, I recommend forking/cloning the repo and creating/submitting the PR from your own branch. It's not strictly required. You can start with a "patch" branch, then set the repo to that branch and keep working on and committing additional files to the branch in the GH UI until the PR is ready. If you go that way tho, I recommend marking the title with "[WIP]" when you commit the first file on the patch PR to let us know you're still working on it.

Start your PR with the text `Fixes #7638`, and this issue will auto-close when the PR merges. I'll call for engineering review after you submit the PR. When they're done, we'll just touch up any MS style guidance issues.

MS style and grammar guidelines are at :point_right: https://docs.microsoft.com/style-guide/welcome/

... and if it isn't covered there, we go by the Chicago Manual of Style .... or we just make it up! :smile: lol We do have quite a few little rules and customs. Don't sweat it tho, we'll touch up little things at the end before merging.

Thanks for the help.

**Comment by granthoff1107** [comment]
I've created a pull request for here https://github.com/aspnet/Docs/pull/7640
